src/functions.php

```diff
@@ -80,15 +80,49 @@ function await(PromiseInterface $promise): mixed
 {
     $fiber = FiberFactory::create();
 
+    $resolved = false;
+    $rejected = false;
+    $resolvedValue = null;
+    $rejectedThrowable = null;
+    $useFiber = false;
```

Comment:
Should we still create the `$fiber` instance in this case?

Reply:
We shouldn't :thinking:

Reply:
Refactored this a bit